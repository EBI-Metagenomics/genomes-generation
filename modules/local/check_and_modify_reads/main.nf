process CHECK_AND_MODIFY_READS {

    label 'process_medium'
    tag "${meta.id}"

    container 'quay.io/biocontainers/seqkit:2.7.0--h9ee0642_0'

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*sanitized*.*.gz"), emit: modified_reads
    path "versions.yml"                      , emit: versions

    script:
    input_ch = reads.collect()
    if (meta.single_end ) {
        """
        seqkit sana ${input_ch[0]} | seqkit replace -p ${meta.id} -r ${meta.assembly_accession} -o ${meta.id}_sanitized.fastq.gz

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            seqkit: \$(seqkit version 2>&1 | sed 's/seqkit //g')
        END_VERSIONS
        """
    }
    else {
        """
        echo "read1"
        seqkit sana ${input_ch[0]} | seqkit replace -p ${meta.id} -r ${meta.assembly_accession} -o ${meta.id}_sanitized_1.fastq.gz

        echo "read2"
        seqkit sana ${input_ch[1]} | seqkit replace -p ${meta.id} -r ${meta.assembly_accession} -o ${meta.id}_sanitized_2.fastq.gz

        echo "sync read1 and read2"
        seqkit pair -1 ${meta.id}_sanitized_1.fastq.gz -2 ${meta.id}_sanitized_2.fastq.gz -O result

        mv result/${meta.id}_sanitized_1.fastq.gz ${meta.id}_sanitized_1.fastq.gz
        mv result/${meta.id}_sanitized_2.fastq.gz ${meta.id}_sanitized_2.fastq.gz
        rm -rf result

        echo "Done"

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            seqkit: \$(seqkit version 2>&1 | sed 's/seqkit //g')
        END_VERSIONS
        """
    }
}