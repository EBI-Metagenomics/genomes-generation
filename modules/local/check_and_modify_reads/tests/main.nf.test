nextflow_process {

    name "Test Process CHECK_AND_MODIFY_READS"
    script "modules/local/check_and_modify_reads/main.nf"
    process "CHECK_AND_MODIFY_READS"

    test("Should run without failures") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = [
                    [id:'SRR3960572', assembly_accession:'ERZ505209'],  // meta
                    [ 
                        file("${moduleDir}/tests/test_1.fq.gz", checkIfExists: true), 
                        file("${moduleDir}/tests/test_2.fq.gz", checkIfExists: true)
                    ]
                ]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
