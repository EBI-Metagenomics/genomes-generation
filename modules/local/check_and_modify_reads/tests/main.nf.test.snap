{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRR3960572",
                            "assembly_accession": "ERZ505209"
                        },
                        [
                            "SRR3960572_sanitized_1.fastq.gz:md5,7f451611c96c9f7a6f3d02fd738fcba3",
                            "SRR3960572_sanitized_2.fastq.gz:md5,f32e13bbb68de431cbb22167db434bdf"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ba8fc5e4d803f79d878a30606ac3a8a8"
                ],
                "modified_reads": [
                    [
                        {
                            "id": "SRR3960572",
                            "assembly_accession": "ERZ505209"
                        },
                        [
                            "SRR3960572_sanitized_1.fastq.gz:md5,7f451611c96c9f7a6f3d02fd738fcba3",
                            "SRR3960572_sanitized_2.fastq.gz:md5,f32e13bbb68de431cbb22167db434bdf"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ba8fc5e4d803f79d878a30606ac3a8a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.01.0"
        },
        "timestamp": "2025-06-11T12:59:13.721499"
    }
}